### PR TITLE
(HP-1106): read license from kubernetes secret

### PR DESCRIPTION
**Pull request by george42-ctds**
Jira Ticket: [HP-1106](https://ctds-planx.atlassian.net/browse/HP-1106)

### Improvements

* Read license secret from kubernetes.

### Deployment changes

[HP-1106]: https://ctds-planx.atlassian.net/browse/HP-1106?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: kube/services/jobs/distribute-licenses-job.yaml [hunk 1]
```diff
@@ -96,15 +98,17 @@ spec:
...
-            # Todo: store, mount licenses secret
-            license_file = """
-            501709301583!$n1d p$53 zvqe 2sfz jzft 7aei e8yL 8ue$ j38b!snic!first line!second line!2100!
-            501709301583!$n1d p$53 zvqe 2sfz jzft 7aei e8yL 8ue$ j38b!snic!first line!second line!2100!
-            501709301583!$n1d p$53 zvqe 2sfz jzft 7aei e8yL 8ue$ j38b!snic!first line!second line!2100!
-            """.strip()
+            # Get license from kubernetes secret
+            print("Ready to read secret")
+            secret_name = os.environ['GEN3_LICENSE_SECRET_NAME']
+            secret_key ="licenseSecrets"
```
Comment (mfshao) [inline kube/services/jobs/distribute-licenses-job.yaml:104]: this value seems to be not being used

File: kube/services/jobs/distribute-licenses-job.yaml [hunk 2]
```diff
@@ -96,15 +98,17 @@ spec:
...
+            license_secrets = os.popen(
+              f"kubectl get secret {secret_name} --template={{{{.data.licenseSecrets}}}} | base64 -d"
```
Comment (mfshao) [inline kube/services/jobs/distribute-licenses-job.yaml:106]: maybe add some comment in this file about how would the license secret looks like so we can have a better idea about how to create them

**Review by mfshao — COMMENTED** [review]
overall LGTM, some comments

File: kube/services/jobs/distribute-licenses-job.yaml (hunk 2 above)
Comment (george42-ctds) [reply kube/services/jobs/distribute-licenses-job.yaml]: Description of secrets has been added in comments. The secret key now comes from env.

**Review by mfshao — APPROVED** [review]

**Pull request merged**